pkg/tsdb/azuremonitor/metrics/url-builder_test.go

```diff
@@ -16,7 +16,7 @@ func TestURLBuilder(t *testing.T) {
 				ResourceName:        "rn",
 			}
 
-			url := ub.Build()
+			url := ub.BuildMetricsURL()
```

Comment:
I'm reviewing these other tests that we didn't touch much and I wonder if we should also have them for resourceURI or if they are not applicable?

I don't have as much historical context for example as to what a short form metric definition is and what the opposite of that is? Maybe we should investigate and if won't apply to resourceUris then we can somehow label these as tests for our legacy handling or something?

Reply:
I think it refers to how the nesting works for this. In the "long form" the metric definition can be something like `Microsoft.NetApp/netAppAccounts/capacityPools/volumes`, and the resource name can be something like `rn1/rn2/rn3`. The metric definition has 4 parts and the resource name has 3 parts delimited by `'/'`. The final 3 parts of the metric definition map to the 3 parts of the resource name so `netAppAccounts/rn1`, then `capacityPools/rn2` and so on.

I think the short form is when there's only one of these pairs. I think it only applies to the legacy handling.

I've just nested these tests without a `ResourceURI` as a legacy test for now and moved the test for the new way to the top.